project/search/rest/views/book.py

```diff
@@ -19,10 +19,21 @@ def get_queryset(self):
         keyword = self.request.query_params.get("keyword", None)
         queryset = super().get_queryset()
         if keyword:
-            queryset = queryset.filter(
-                Q(book__author__name__icontains=keyword.lower())
-                | Q(book__name__icontains=keyword.lower())
-                | Q(book__publisher__name__icontains=keyword.lower())
-            ).order_by("book__author__name")
+            queryset = (
+                queryset.filter(
+                    Q(book__author__name__icontains=keyword.lower())
+                    | Q(book__name__icontains=keyword.lower())
+                    | Q(book__publisher__name__icontains=keyword.lower())
+                ).annotate(
+                    relevance=Case(
+                        When(book__name__icontains=keyword.lower(), then=Value(3)),
```

Comment:
what if we use the same point and then check which on has a higher score the we order by value